### PR TITLE
[NEW FEATURE] Add addons to available variables

### PR DESCRIPTION
**Pull request by l-j**
I think it will be beneficial to add to this component a possibility to list not only HACS components, but Home Assistant add-ons also. That way it will be possible for a user of the plugin to have a list of all extensions (HACS Integrations, plugins, themes and Home Assistant Add-ons) in one place, for example as a part of the documentation for Home Assistant configuration (this is how I am using it).

In this PR I updated README and the default template as well, so it will be more clear for others, what are the capabilities of this component.

Tested on Home Assistant 2021.12.8, HACS 0.19.2.

### CODE REVIEW

**Review by shiner66 — APPROVED** [review]

**Review by shiner66 — APPROVED** [review]
Works correctly on my installation

**Comment by l-j** [comment]
@ludeeus Can you take a look?

File: custom_components/readme/__init__.py [hunk 1]
```diff
@@ -185,6 +187,15 @@ def get_hacs_components(hass: HomeAssistant):
     ]
 
 
+def get_ha_installed_addons(hass: HomeAssistant) -> List[Dict[str, Any]]:
+    supervisor_info = hass.components.hassio.get_supervisor_info()
+
```
Comment (ludeeus) [inline custom_components/readme/__init__.py:192]:
```suggestion
    if not hass.components.hassio.is_hassio():
        return []
    supervisor_info = hass.components.hassio.get_supervisor_info()

```

File: custom_components/readme/__init__.py [hunk 2]
```diff
@@ -185,6 +187,15 @@ def get_hacs_components(hass: HomeAssistant):
     ]
 
 
+def get_ha_installed_addons(hass: HomeAssistant) -> List[Dict[str, Any]]:
```
Comment (ludeeus) [inline custom_components/readme/__init__.py:190]: Import and decorate this with `callback` from `homeassistant.core`

File: custom_components/readme/default.j2 [hunk 3]
```diff
@@ -11,16 +10,29 @@ Number of entities | {{states | count}}
...
-{% endfor -%}
-{% endif %}
+## My installed extensions:
+
+### Add-ons
+{%- for addon in addons | sort(attribute='name') %}
+- {{addon.name}}
+{%- endfor %}
+
+### HACS Integrations (custom_components)
```
Comment (ludeeus) [inline custom_components/readme/default.j2:20]:
```suggestion
### Custom integrations
```

File: custom_components/readme/default.j2 [hunk 4]
```diff
@@ -11,16 +10,29 @@ Number of entities | {{states | count}}
...
+{%- for component in hacs_components | selectattr('category', 'equalto', 'integration') | sort(attribute='name') %}
+- [{{component.name}}]({{component.documentation}})
+{%- endfor %}
+
+### HACS Frontend - Plugins
```
Comment (ludeeus) [inline custom_components/readme/default.j2:25]:
```suggestion
### Lovelace plugins
```

File: custom_components/readme/default.j2 [hunk 5]
```diff
@@ -11,16 +10,29 @@ Number of entities | {{states | count}}
...
+{%- for component in hacs_components | selectattr('category', 'equalto', 'plugin') | sort(attribute='name') %}
+- [{{component.name}}]({{component.documentation}})
+{%- endfor %}
+
+### HACS Frontend - Themes
```
Comment (ludeeus) [inline custom_components/readme/default.j2:30]:
```suggestion
### Themes
```

File: custom_components/readme/__init__.py [hunk 6]
```diff
@@ -185,6 +187,15 @@ def get_hacs_components(hass: HomeAssistant):
...
+    supervisor_info = hass.components.hassio.get_supervisor_info()
+
+    if supervisor_info:
+        return supervisor_info.get("addons", [])
+    else:
+        return []
```
Comment (ludeeus) [inline custom_components/readme/__init__.py:196]:
```suggestion
    return []
```

File: custom_components/readme/__init__.py [hunk 7]
```diff
@@ -10,7 +10,7 @@
 import json
 import os
 from shutil import copyfile
-from typing import Any, List
+from typing import Any, List, Dict
```
Comment (ludeeus) [inline custom_components/readme/__init__.py:13]:
```suggestion
from typing import Any, Dict, List
```

File: README.md [hunk 8]
```diff
@@ -120,6 +131,29 @@ _{{custom_component_descriptions[integration.domain]}}_
 {%- endfor %}
 ```
 
+### addons
```
Comment (ludeeus) [inline README.md:134]:
```suggestion
### Add-ons

```

File: README.md [hunk 9]
```diff
@@ -11,18 +11,24 @@
...
 ## Installation
 
-1. Using the tool of choice open the directory (folder) for your HA configuration (where you find `configuration.yaml`).
-2. If you do not have a `custom_components` directory (folder) there, you need to create it.
-3. In the `custom_components` directory (folder) create a new folder called `readme`.
-4. Download _all_ the files from the `custom_components/readme/` directory (folder) in this repository.
-5. Place the files you downloaded in the new directory (folder) you created.
-6. Restart Home Assistant
-7. Choose:
+1. Install it with HACS
```
Comment (ludeeus) [inline README.md:20]:
```suggestion
1. Download it with HACS

```

File: README.md [hunk 10]
```diff
@@ -11,18 +11,24 @@
...
+OR 
+
+1. Install it manually
+   1. Using the tool of choice open the directory (folder) for your HA configuration (where you find `configuration.yaml`).
+   2. If you do not have a `custom_components` directory (folder) there, you need to create it.
+   3. In the `custom_components` directory (folder) create a new folder called `readme`.
+   4. Download _all_ the files from the `custom_components/readme/` directory (folder) in this repository.
+   5. Place the files you downloaded in the new directory (folder) you created.
+2. Restart Home Assistant
+3. Choose:
```
Comment (ludeeus) [inline README.md:31]:
```suggestion
2. Restart Home Assistant
3. Choose:
```

**Review by ludeeus — CHANGES_REQUESTED** [review]

File: custom_components/readme/__init__.py (hunk 2 above)
Comment (l-j) [reply custom_components/readme/__init__.py:191]: done

**Comment by l-j** [comment]
I have tested changes again against Home Assistant 2022.2.9 and HACS 1.22.0

**Comment by l-j** [comment]
@ludeeus can you re-check?

**Review by ludeeus — APPROVED** [review]

**Pull request merged**